cove_ofds/process.py

```diff
@@ -0,0 +1,321 @@
+        self.data_filename = os.path.join(
+            self.supplied_data.data_dir(), "data_from_geojson.json"
+        )
+
+    def process(self, process_data: dict) -> dict:
+        if self.supplied_data.format != "geojson":
+            return process_data
+
+        # check already done
+        if os.path.exists(self.data_filename):
+            process_data["json_data_filename"] = self.data_filename
+            return process_data
+
+        # Get files
+        supplied_data_json_files = SuppliedDataFile.objects.filter(
+            supplied_data=self.supplied_data
+        )
+        nodes_data_json_files = [
+            f for f in supplied_data_json_files if f.meta.get("geojson") == "nodes"
+        ]
+        spans_data_json_files = [
+            f for f in supplied_data_json_files if f.meta.get("geojson") == "spans"
+        ]
+
+        if len(nodes_data_json_files) != 1 or len(spans_data_json_files) != 1:
+            raise Exception("Can't find JSON original data!")
+
+        # Get data from files
+        nodes_data_json_file = nodes_data_json_files[0]
+        spans_data_json_file = spans_data_json_files[0]
+
+        with open(nodes_data_json_file.upload_dir_and_filename()) as fp:
+            nodes_data = json.load(fp)
+
+        with open(spans_data_json_file.upload_dir_and_filename()) as fp:
+            spans_data = json.load(fp)
+
+        # Convert
+        converter = GeoJSONToJSONConverter()
+        converter.process_data(nodes_data, spans_data)
+
+        # Save
+        with open(self.data_filename, "w") as fp:
+            json.dump(converter.get_json(), fp, indent=4)
+
+        # Info
+        process_data["json_data_filename"] = self.data_filename
+        return process_data
+
+    def get_context(self):
+        # Info
+        context = {}
+        # original format
+        if self.supplied_data.format == "geojson":
+            context["original_format"] = "geojson"
+            # Download data
+            if os.path.exists(self.data_filename):
+                context["can_download_json"] = True
+                context["download_json_url"] = os.path.join(
+                    self.supplied_data.data_url(), "data_from_geojson.json"
+                )
+                context["download_json_size"] = os.stat(self.data_filename).st_size
+            else:
+                context["can_download_json"] = False
+        # Return
+        return context
+
+
+class ConvertJSONIntoGeoJSON(ProcessDataTask):
+    """Convert primary format (JSON) to GeoJSON"""
+
+    def __init__(self, supplied_data):
+        super().__init__(supplied_data)
+        self.nodes_file_name = os.path.join(
+            self.supplied_data.data_dir(), "nodes.geo.json"
+        )
+        self.spans_file_name = os.path.join(
+            self.supplied_data.data_dir(), "spans.geo.json"
+        )
+
+    def process(self, process_data: dict) -> dict:
+        # TODO if original format this, don't bother
```

Comment:
It's not clear to me what this comment means. Should it be?

Reply:
"if user uploaded GeoJSON, don't bother to convert to GeoJSON" - I'll sort later